model/task/db.go

```diff
@@ -65,6 +65,7 @@ var (
 	LatestParentExecutionKey       = bsonutil.MustHaveTag(Task{}, "LatestParentExecution")
 	OldTaskIdKey                   = bsonutil.MustHaveTag(Task{}, "OldTaskId")
 	ArchivedKey                    = bsonutil.MustHaveTag(Task{}, "Archived")
+	CanResetKey                    = bsonutil.MustHaveTag(Task{}, "CanReset")
```

Comment:
`CanReset` is not a proxy for having a completed status but rather is meant to be a protection mechanism against a situation where we succeed archiving but fail to reset the task, so we can still have a way to retry the "restart" operation to fix the state.  I'm realizing now that setting `CanReset` to true on `MarkEnd` and `MarkSystemFailed` defeats the benefit of having it - we should only set it directly after a successful archival. I've removed those instances.

Since it can now only ever get toggled true upon successful archival, it shouldn't be possible to have `CanReset` true for a task in an unfinished state.  Likewise the converse shouldn't be possible. `CanReset` should just be ephemerally set to true for a split second between archiving and resetting, and nothing else.

Reply:
Ah yeah, this usage of `CanReset` makes more sense now. Thanks!